runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/net/DefaultHostResolverJVM.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.net
+
+import aws.smithy.kotlin.runtime.util.InternalApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.net.Inet4Address
+import java.net.Inet6Address
+import java.net.InetAddress
+
+internal actual object DefaultHostResolver : HostResolver {
+    override suspend fun resolve(hostname: String): List<HostAddress> = withContext(Dispatchers.IO) {
+        InetAddress.getAllByName(hostname).map { it.toHostAddress() }
+    }
+    override fun reportFailure(addr: HostAddress) { }
```

Comment:
**question**: Should there be a default caching behavior? If so what?

Reply:
I think there needs to be higher-level caching to meet the spirit of the reference architecture. `InetAddress` provides some caching but without the ability to invalidate or purge. If we're going to use that, it implies to me that we resolve what it reports (post-low-level caching) with what we know from invalidations and purges...or bypass `InetAddress` altogether and _still_ write our own caching layer.